### PR TITLE
[WIP] add links to academy

### PR DESCRIPTION
**Pull request by rosieyohannan**
Added links to the following academy courses in the relevant docs:

- [x] runner
- [x] dynamic config
- [x] cicd 101
- [x] insights
- [x] migrations
- [x] DLC
- [x] build envs
- [x] orbs
- [x] arm

Also - reinstated the executor intro doc into the navigation and ensured we make reference to "build environments" in the executor types and intro docs to maintain consistency with the language we use in academy etc.

### CODE REVIEW

File: jekyll/_cci2/dynamic-config.md [hunk 1]
```diff
@@ -88,3 +88,7 @@ documentation for more information.
   - [Execute specific `workflows` or `steps` based on which files are modified]({{ site.baseurl }}/2.0/configuration-cookbook/?section=examples-and-guides#execute-specific-workflows-or-steps-based-on-which-files-are-modified)
 - The [`continuation`](https://circleci.com/developer/orbs/orb/circleci/continuation) orb
 - The [`continuePipeline`](https://circleci.com/docs/api/v2/#operation/continuePipeline) API call
+
+## Learn More
+{: #learn-more }
+Take the [dynamic configuration course](https://academy.circleci.com/dynamic-config) to learn more.
```
Comment (rosieyohannan) [inline jekyll/_cci2/dynamic-config.md:94]:
```suggestion
Take the [dynamic configuration course](https://academy.circleci.com/dynamic-config) with CircleCI academy to learn more.
```

File: jekyll/_cci2/runner-overview.adoc [hunk 2]
```diff
@@ -141,3 +141,9 @@ Almost all standard CircleCI features are available for use with runner jobs, bu
 - https://circleci.canny.io/runner-feature-requests/p/support-rerun-with-ssh-on-runner[Rerun with SSH]
 - https://circleci.canny.io/runner-feature-requests/p/support-test-splitting-on-self-hosted-runners[Test Splitting]
 - https://circleci.canny.io/runner-feature-requests/p/support-addsshkey-on-self-hosted-runners[`add_ssh_keys`]
+
+== Learn more
+
+Take the https://academy.circleci.com/runner-course[runner course] with CircleCI academy to learn more about running jobs on your infrastructure.
```
Comment (thomasr8) [inline jekyll/_cci2/runner-overview.adoc:147]:
```suggestion
Take the https://academy.circleci.com/runner-course[runner course] with CircleCI Academy to learn more about running jobs on your infrastructure.
```

**Review by thomasr8 — APPROVED** [review]
Looks good @rosieyohannan . I capitalized instances of "academy" to "Academy," but I am not sure if we should capitalize it or not. Thoughts?

**Comment by rosieyohannan** [comment]
> Looks good @rosieyohannan . I capitalized instances of "academy" to "Academy," but I am not sure if we should capitalize it or not. Thoughts?

Yes I think you're right :-)

**Comment by rosieyohannan** [comment]
Also I was confused by your change to "enroll in" and I found this! There are more differences between US and UK english than I thought!

Screenshot 2021-08-24 at 11 00 42

**Comment by rosieyohannan** [comment]
Thanks for the review @thomasr8

**Pull request merged**